lib/utils.js

```diff
@@ -50,25 +50,36 @@ function retryDelay(attempt) {
 }
 
 /**
- * Returns a google auth client for the current application and provided
- * scopes using a cached client if one exists.
+ * Returns a google auth client for the current application with provided
+ * scopes and configuration, using a cached client if one exists.
  */
-function getApplicationDefaultAuth(scopes, callback) {
-  scopes.sort();
-  if (cachedAuthClients[scopes]) {
-    callback(null, cachedAuthClients[scopes]);
+function getAuthClient(scopes, config, callback) {
+  if (isFunction(config)) {
```

Comment:
We should come up with a general stance on using `lodash` to typecheck. I'm personally not as big a fan of bringing in extra dependencies but we do it in some of our agents. For reference, `isFunction` only differs in behavior from a `typeof` check on Safari 9 due to a bug in Safari as far as I can tell.